app/routes/api.py

```
@@ -806,30 +544,60 @@ def approve_redemption():
+                    event_type="CONSUMED",
+                    correlation_id=pending_action.correlation_id,
+                    payload={
+                        "outcome": "APPROVED",
+                        "source": "api.approve_redemption",
+                        "item_type": store_item.item_type,
+                        "details": (pending_action.payload or {}).get("details") or None,
+                    },
+                )
+            )
+        db.session.delete(pending_action)
+    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)